File: BookExamples/39_LifecycleDemo/app/src/main/java/com/ebookfrenzy/lifecycledemo/ui/main/MainFragment.java
package com.ebookfrenzy.lifecycledemo.ui.main;

//older class
//import androidx.lifecycle.ViewModelProviders;
import androidx.lifecycle.ViewModelProvider;


import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.ebookfrenzy.lifecycledemo.R;
import com.ebookfrenzy.lifecycledemo.DemoObserver;
import com.ebookfrenzy.lifecycledemo.DemoOwner;

public class MainFragment extends Fragment {

    private MainViewModel mViewModel;
    private DemoOwner demoOwner;

    public static MainFragment newInstance() {
        return new MainFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.main_fragment, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        //book had this
        //mViewModel = ViewModelProviders.of(this).get(MainViewModel.class);

        //replaced with this
        mViewModel = new ViewModelProvider(this).get(MainViewModel.class);

        // TODO: Use the ViewModel
        //demoOwner = new DemoOwner();
        //demoOwner.startOwner();
        //demoOwner.stopOwner();
        getLifecycle().addObserver(new DemoObserver());
    }

}

File: class_book_examples/pre_eel/chapter46_LifecycleDemo/app/src/main/java/com/ebookfrenzy/lifecycledemo/ui/main/MainFragment.kt
package com.ebookfrenzy.lifecycledemo.ui.main

import androidx.lifecycle.ViewModelProvider
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.ebookfrenzy.lifecycledemo.R

import com.ebookfrenzy.lifecycledemo.DemoObserver
import com.ebookfrenzy.lifecycledemo.DemoOwner

class MainFragment : Fragment() {

    companion object {
        fun newInstance() = MainFragment()
    }

    private lateinit var viewModel: MainViewModel

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return inflater.inflate(R.layout.main_fragment, container, false)
    }


    //private lateinit var demoOwner: DemoOwner

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)


        lifecycle.addObserver(DemoObserver())

        //THESE METHODS ARE FOR THE CUSTOM CLASS.
        //demoOwner = DemoOwner()
        //demoOwner.startOwner()
        //demoOwner.stopOwner()

    }

}